service/history/api/queryworkflow/api.go

```diff
@@ -123,6 +123,12 @@ func Invoke(
 		return nil, consts.ErrWorkflowTaskNotScheduled
 	}
 
+	if _, status := mutableState.GetWorkflowStateStatus(); status == enumspb.WORKFLOW_EXECUTION_STATUS_TERMINATED &&
```

Comment:
Timeout or other closed status as well. Should check status != WORKFLOW_EXECUTION_STATUS_RUNNING

Reply:
Done

Reply:
Use mutableState.IsWorkflowExecutionRunning() ?

Reply:
IsWorkflowExecutionRunning() checks `ms.executionState.State`. 
Here I am checking `ms.executionState.Status`. 
is checking `WORKFLOW_EXECUTION_STATE_RUNNING` or `WORKFLOW_EXECUTION_STATE_RUNNING` the same thing in this case?

Reply:
It makes sense to use mutableState.IsWorkflowExecutionRunning(). Updated.